python/TestHarness/schedulers/RunParallel.py

```diff
@@ -0,0 +1,130 @@
+
+        return params
+
+    def __init__(self, harness, params):
+        Scheduler.__init__(self, harness, params)
+
+    ## Run the test!
+    ## Note: use thread_lock where necessary
+    #
+    #  with thread_lock:
+    #      do protected stuff
+    #
+    # The run method should be blocking until the test has completed _and_ the results have been
+    # processed (processResults()). When we return from this method, this test will be considered
+    # finished and added to the status pool for printing out the final results.
+    def run(self, tester, testers, thread_lock, options):
+        # Ask the tester if its allowed to run on this machine
+        if tester.checkRunnableBase(options):
+
+            # Ask if this test is allowed to run based on other test's circumstances
+            status_check = StatusDependency(tester, testers, options)
+            if status_check.checkAndSetStatus():
+                # Get the command needed to run this test
+                command = tester.getCommand(options)
+
+                # Prepare to run the test
+                tester.prepare(options)
+
+                # If shouldExecute was false, there is process or output to work with so pass
+                # control over to testOutput now, to run processResults and complete the test
+                if not tester.shouldExecute() or options.dry_run:
+                    tester.start_time = clock()
+                    tester.exit_code = 0
+                    tester.end_time = clock()
+                    tester.std_out = ''
+                    self.testOutput(tester, testers, '', options)
+                    return
+                else:
+                    # Launch the command and start the clock
+                    (process, output_file, start_time) = returnCommand(tester, command)
+
+                # Set the tester's start time
+                tester.start_time = start_time
+
+                # We're a thread so wait for the process to complete
+                process.wait()
+                tester.end_time = clock()
+
+                # Did the process fail and we didn't know it (someone told the testharness we timed out)?
+                if tester.didFail():
+                    return
+
+                if process.poll() is not None:
+                    # set the exit code
+                    tester.exit_code = process.poll()
+
+                    # set the tester output (trimmed)
+                    results = readOutput(output_file, options)
+                    output_file.close()
+
+                    # Process the results and beautify the output
+                    self.testOutput(tester, testers, results, options)
+
+                # This test failed to launch properly
+                else:
+                    tester.setStatus('ERROR LAUNCHING JOB', tester.bucket_fail)
+
+        else:
+            tester.end_time = clock()
+
+    def getDependents(self, tester, testers):
```

Comment:
I don't think this method has the right name. It's called `getDependents` but it returns a Boolean.